common/sgx/eeid.c

```diff
@@ -0,0 +1,451 @@
+        sizeof(eeid->size_settings)));
+    OE_CHECK(serialize_elem(
+        "data_size",
+        p,
+        &r,
+        (uint8_t*)&eeid->data_size,
+        sizeof(eeid->data_size)));
+    OE_CHECK(serialize_elem(
+        "vaddr", p, &r, (uint8_t*)&eeid->vaddr, sizeof(eeid->vaddr)));
+    OE_CHECK(serialize_elem(
+        "entry_point",
+        p,
+        &r,
+        (uint8_t*)&eeid->entry_point,
+        sizeof(eeid->entry_point)));
+    OE_CHECK(
+        serialize_elem("data", p, &r, (uint8_t*)eeid->data, eeid->data_size));
+
+    **p = '\0';
+
+done:
+    return OE_OK;
+}
+
+oe_result_t oe_deserialize_eeid(
+    const char* buf,
+    size_t buf_size,
+    oe_eeid_t* eeid)
+{
+    oe_result_t result;
+
+    if (!buf || !buf_size || !eeid)
+        return OE_INVALID_PARAMETER;
+
+    const char** p = &buf;
+    size_t r = buf_size;
+
+    memset(eeid, 0, sizeof(oe_eeid_t));
+
+    OE_CHECK(deserialize_elem(
+        p, &r, (uint8_t*)eeid->hash_state.H, sizeof(eeid->hash_state.H)));
+    OE_CHECK(deserialize_elem(
+        p, &r, (uint8_t*)eeid->hash_state.N, sizeof(eeid->hash_state.N)));
+    OE_CHECK(deserialize_elem(
+        p, &r, (uint8_t*)&eeid->signature_size, sizeof(eeid->signature_size)));
+    OE_CHECK(deserialize_elem(p, &r, eeid->signature, eeid->signature_size));
+    OE_CHECK(deserialize_elem(
+        p, &r, (uint8_t*)&eeid->size_settings, sizeof(eeid->size_settings)));
+    OE_CHECK(deserialize_elem(
+        p, &r, (uint8_t*)&eeid->data_size, sizeof(eeid->data_size)));
+    OE_CHECK(
+        deserialize_elem(p, &r, (uint8_t*)&eeid->vaddr, sizeof(eeid->vaddr)));
+    OE_CHECK(deserialize_elem(
+        p, &r, (uint8_t*)&eeid->entry_point, sizeof(eeid->entry_point)));
+    OE_CHECK(deserialize_elem(p, &r, (uint8_t*)eeid->data, eeid->data_size));
+
+done:
+    return OE_OK;
+}
+
+oe_result_t oe_replay_eeid_pages(
+    const oe_eeid_t* eeid,
+    struct _OE_SHA256* cpt_mrenclave,
+    bool with_eeid_pages)
+{
+    oe_result_t result;
+    oe_sha256_context_t hctx;
+    oe_sha256_restore(&hctx, eeid->hash_state.H, eeid->hash_state.N);
+
+    uint64_t base = 0x0ab0c0d0e0f;
+    oe_page_t blank_pg, stack_pg, tcs_pg;
+    memset(&blank_pg, 0, sizeof(blank_pg));
+    memset(&stack_pg, 0xcc, sizeof(stack_pg));
+
+#define ADD_PAGE(PG, T)                                      \
+    {                                                        \
+        OE_CHECK(oe_sgx_measure_load_enclave_data(           \
+            &hctx,                                           \
+            base,                                            \
+            base + vaddr,                                    \
+            (uint64_t)&PG,                                   \
+            SGX_SECINFO_REG | SGX_SECINFO_R | SGX_SECINFO_W, \
+            T));                                             \
+        vaddr += OE_PAGE_SIZE;                               \
+    }
+
+    uint64_t vaddr = eeid->vaddr;
+
+    if (with_eeid_pages)
```

Comment:
This is where we replay the addition of memory pages, both, for verification of the extended image hash (`with_eeid_pages=true`) and the base image hash, for which there are no EEID pages, but one TCS page, or more if we decide to support arbitrary base image properties.

Reply:
Rather than commenting in a github comment, I'd recommend the comment be inserted into the actual source code.

Reply:
Done.